app/models/slot_availability.rb

```diff
@@ -1,12 +1,32 @@
 class SlotAvailability
   attr_reader :slots
 
-  def initialize(prison:)
+  def initialize(prison:, use_nomis_slots: false)
     @prison = prison
-    @slots = prison.available_slots.to_a
+    @slots = lookup_slots_for_prison(prison, use_nomis_slots)
   end
 
   # rubocop:disable Metrics/MethodLength
+  def lookup_slots_for_prison(prison, use_nomis)
```

Comment:
There is a lot of branching going on this method, I'd find it clearer if it was something like:

```ruby
def initialize(prison:, use_nomis_slots: false)
  @prison = prison
  @use_nomis = use_nomis_lots && Nomis::Api.enabled?
  @slots = lookup_nomis_slots(prison) || prison.available_slots.to_a
end

def lookup_nomis_slots(prison)
  return nil unless @use_nomis

  begin
     Nomis::Api.instance.fetch_bookable_slots(
       prison: prison,
      start_date: prison.first_bookable_date,
      end_date: prison.last_bookable_date
    )
 rescue Excon::Errors::Error => e
   Rails.logger.warn "Error calling the NOMIS API: #{e.inspect}"
   nil
end
```

With something like this we also avoid having to put the default slots in the rescue and the else branch.

@tatyree @mloughran what do you think?